src/utils/system_utils/tests/system_utils_ut.cpp

```diff
@@ -9,81 +9,214 @@
 using Catch::Matchers::Equals;
 
 #include "aduc/system_utils.h"
-
 #include <aduc/auto_opendir.hpp>
-#include <algorithm> // std::find*
-#include <dirent.h>
-#include <sstream>
 #include <sys/stat.h>
 #include <vector>
 
-static bool callback_called = false;
-std::string actual;
+/**
+ * @brief A capture of the arguments passed in a call invocation to callback for each sub dir in base dir.
+ *
+ */
+struct ForEachDirCallRecord
+{
+    void* context;
+    std::string baseDir;
+    std::string subDir;
+};
+
+/**
+ * @brief ForEachDirCallRecord equals operator needed for STL container.
+ *
+ * @param first The first call record object.
+ * @param second The second call record object.
 
-static void perDirFn(void* context, const char* baseDir, const char* subdir_name)
+ * @return true when equal.
+ */
+bool operator==(const ForEachDirCallRecord& first, const ForEachDirCallRecord& second)
 {
-    callback_called = true;
-    std::stringstream ss;
+    return first.context == second.context && first.baseDir == second.baseDir && first.subDir == second.subDir;
```

Comment:
comparing context pointers? wouldn't you want to compare the context objects?

Reply:
Normally yes, but for the purposes of the tests it just wants to verify that the obj ptr of the recorded call matches what was expected.